Cpp_test/lib/ipc-shm-receive.cpp

```diff
@@ -0,0 +1,110 @@
+#include "ipc-shm-receive.h"
+
+IPCShmReceive::~IPCShmReceive()
+{
+
+	shm_unlink(this->info.method_name);
+	if (pthread_cond_destroy(&(this->shm_ptr->cond_re)) != 0)
+	{
+		std::cerr << "Error at pthread_cond_destroy(): receive" << std::endl;
+	}
+	if (pthread_cond_destroy(&(this->shm_ptr->cond)) != 0)
+	{
+		std::cerr << "Error at pthread_cond_destroy(): send" << std::endl;
+	}
+	if (pthread_mutex_destroy(&(this->shm_ptr->mutex)) != 0)
+	{
+		std::cerr << "Error at pthread_mutex_destroy()" << std::endl;
+	}
```

Comment:
In this destructor, you should get the result of the functions to print the result if there is an issue, like:
```c++
int ret = pthread_cond_destroy(&(this->shm_ptr->cond)) ;
if (ret != 0) {
   std::cerr << "Error at pthread_cond_destroy(): " << strerror(ret) << std::endl;
}
```

Reply:
If I am running only the receiver I have a segmentation fault, I think you should be sure that mutex attr/ mutex and condvar are initialized before destroying them.
![image](https://user-images.githubusercontent.com/97879298/208362466-6929ad9d-07b2-46b5-bdc8-53227b7be510.png)